src/TapeChart.js

```diff
@@ -0,0 +1,48 @@
+class TapeChart {
+  constructor(users, rooms, bookings) {
+    this.users = users;
+    this.rooms = rooms;
+    this.bookings = bookings;
+  }
+
+getAvailableRoomsByDate(date) {
+    let availableByDate = this.rooms.reduce((acc, room) => {
+      let bookingsByRoom = this.bookings.filter(booking =>
+        booking.roomNumber === room.number);
+      if (!bookingsByRoom.some(booking => booking.date === date)) {
+        acc.push(room);
+      }
+        return acc;
+    }, [])
+  return availableByDate;
+}
+
+getOccupancy(date) {
+  return 100 - (this.getAvailableRoomsByDate(date).length
+    / this.rooms.length * 100);
+}
+
+getBookingsByDate(date) {
+  return this.bookings.filter(booking => booking.date === date);
+}
+
+getDaysRevenue(date) {
+  let bookingsByDate = this.getBookingsByDate(date);
+  return bookingsByDate.reduce((acc, booking) => {
+    this.rooms.forEach(room => {
+      if (room.number === booking.roomNumber) {
+        acc += room.costPerNight;
+      }
+    })
+    return acc;
+  }, 0)
+}
+
+filterAvailableRoomsByFeature(date, feature, spec) {
```

Comment:
I think this is really good.

Reply:
I initially tried to pass an object with the key:value pair but did not have much success.  This way is a little easier for me to read and manipulate anyway.